sklearn/utils/sparsefuncs.py

```diff
@@ -159,8 +185,6 @@ def incr_mean_variance_axis(X, *, axis, last_mean, last_var, last_n):
         else:
             return _incr_mean_var_axis0(X.T, last_mean=last_mean,
                                         last_var=last_var, last_n=last_n)
-    else:
-        _raise_typeerror(X)
```

Comment:
Why has this been removed?

Reply:
it has been move at the top to fail early